Pages/FoodPages/Reservations.cshtml.cs

```diff
@@ -28,12 +31,29 @@ public async Task OnGetAsync()
             ApplicationUser applicationUser = await _userManager.GetUserAsync(User);
             
             var restaurantsFoods = applicationUser.FoodList;
+
+            if (!string.IsNullOrEmpty(searchString))
+            {
+                cart = getCartFromDataBase(restaurantsFoods, searchString);
+            }
+            else
+            {
+                cart = getCartFromDataBase(restaurantsFoods);
+            }
+        }
+        public IList<ShoppingCart> getCartFromDataBase(IList<Food> restaurantsFoods, string searchString = null)
+        {
             var result = from a in restaurantsFoods
                          join b in _context.Cart.ToList()
                          on a.ID equals b.FoodId
-                         where b.Reserved = true
+                         where b.Reserved = true &&
+                            (searchString is not null && (
+                             b.Customer.FirstName.Contains(searchString) ||
+                             b.Customer.LastName.Contains(searchString) ||
+                             b.Food.FoodName.Contains(searchString)) ||
+                             searchString is null)
```

Comment:
Maybe you could explain a little bit this part You can write in a group chat.

Reply:
Basically, it is just as a filter. I will explain it with more details in the chat